src/shared/utils.ts

```diff
@@ -700,7 +700,7 @@ export function setupTribute() {
         trigger: "!",
         selectTemplate: (item: any) => {
           const it: CommunityTribute = item.original;
-          return `[${it.key}](${it.view.community.actor_id})`;
+          return `${it.key}`;
```

Comment:
I'm not a fan of this approach.

Instead, just have two links, the fedi one, and whatever you local instance is with the community.name.

IE something like `[it.key](community.actor_id),[^local^](this_instance/c/@{community.name}`

Reply:
My worry with having two links is that it doesn't make the linking experience transparent. In my opinion, the user shouldn't need to know about the nuances of instances and how they interact in order to consume content. They should just be able to click a link and get sent to the version of the post on their home instance; i.e. they should be able to get a single link and it should "just work".

Also, having the separate "local" link breaks interoperability with other fediverse services. The motivation behind my approach is that when a user posts they just put text in a particular format, and when that post gets fetched by other services (like Kbin), the other service can handle (or not handle) the formatted text in whichever manner they like.

Reply:
Hmm, would it be a good idea to implement a companion solution to that issue then? That is, should we make it so that if you navigate to a post on a remote instance which hasn't been federated with your home instance yet, the home instance will automatically fetch the content from the remote instance instead of 404'ing? Or was there an explicit design choice to 404 instead of fetching?

Reply:
There aren't any 404s currently, because the post and comment nodes already exist on your own instance, or you wouldn't be seeing them (so neither link will 404).

But with searched content like this, the comment with text referencing a federated community or actor, might not exist everywhere. 

> the home instance will automatically fetch the content from the remote instance instead of 404'ing?

We don't currently do this, but this is the advantage of having a local link. We could do a [resolveObject](https://join-lemmy.org/api/classes/LemmyHttp.html#resolveObject) if it 404s, and cache it to our local instance. No guarantees still, since it could be blocked, but at least its possible.

We've chosen for resolve to only work on the search page currently, and by logged-in users only, but I'm open to a discussion for changing that. Def has pros and cons.

Reply:
Another solution if we want to keep the explicit hyperlinks to different instances could be to implement a "You are leaving your home instance" screen before redirection. That way users are explicitly told that they are moving to a physically different website and they don't have to rely on vague UI clues to realize that.